Kronos-Shifts-Connector/Microsoft.Teams.Shifts.Integration/Microsoft.Teams.Shifts.Integration.API/Controllers/SyncKronosToShiftsController.cs

```diff
@@ -104,51 +104,122 @@ public ActionResult SyncKronosToShifts(string isRequestFromLogicApp)
-
+                isOpenShiftRequestSyncSuccessful = true;
                 this.telemetryClient.TrackTrace($"{Resource.ProcessOpenShiftsRequests} completed from {Resource.ProcessKronosToShiftsShiftsAsync} ");
+            }
+#pragma warning disable CA1031 // Do not catch general exception types
+            catch (Exception ex)
+#pragma warning restore CA1031 // Do not catch general exception types
+            {
+                isOpenShiftRequestSyncSuccessful = false;
+                this.telemetryClient.TrackException(ex);
+            }
 
+             try
+            {
                 // Sync swap shifts from kronos to shifts.
                 await this.swapShiftController.ProcessSwapShiftsAsync(isRequestFromLogicApp).ConfigureAwait(false);
-
+                isSwapShiftRequestSyncSuccessful = true;
                 this.telemetryClient.TrackTrace($"{Resource.ProcessSwapShiftsAsync} completed from {Resource.ProcessKronosToShiftsShiftsAsync} ");
+            }
+#pragma warning disable CA1031 // Do not catch general exception types
+            catch (Exception ex)
+#pragma warning restore CA1031 // Do not catch general exception types
+            {
+                isSwapShiftRequestSyncSuccessful = false;
+                this.telemetryClient.TrackException(ex);
+            }
 
+             try
+            {
                 // Sync timeoffreasons from kronos to shifts.
                 await this.timeOffReasonController.MapPayCodeTimeOffReasonsAsync().ConfigureAwait(false);
-
+                isMapPayCodeTimeOffReasonsSuccessful = true;
                 this.telemetryClient.TrackTrace($"{Resource.MapPayCodeTimeOffReasonsAsync} completed from {Resource.ProcessKronosToShiftsShiftsAsync} ");
+            }
+#pragma warning disable CA1031 // Do not catch general exception types
+            catch (Exception ex)
+#pragma warning restore CA1031 // Do not catch general exception types
+            {
+                isMapPayCodeTimeOffReasonsSuccessful = false;
+                this.telemetryClient.TrackException(ex);
+            }
 
-                // Sync timeoff from kronos to shifts.
-                await this.timeOffController.ProcessTimeOffsAsync(isRequestFromLogicApp).ConfigureAwait(false);
-
-                this.telemetryClient.TrackTrace($"{Resource.ProcessTimeOffsAsync} completed from {Resource.ProcessKronosToShiftsShiftsAsync} ");
-
-                // Sync timeoff from kronos to shifts.
-                await this.timeOffRequestsController.ProcessTimeOffRequetsAsync(isRequestFromLogicApp).ConfigureAwait(false);
-
-                this.telemetryClient.TrackTrace($"{Resource.SyncTimeOffRequestsFromShiftsToKronos} completed from {Resource.ProcessKronosToShiftsShiftsAsync} ");
+            // Sync TimeOff and TimeOffRequest only if Paycodes in kronos synced successfully.
+             if (isMapPayCodeTimeOffReasonsSuccessful)
```

Comment:
Alignment!

Reply:
The alignment in all your added lines is wonky.

Reply:
@aosolis - ack'd on the alignment changes

Reply:
This has been resolved - please see the newest changes